### ISSUE TITLE
[Outdated] neovim-nightly-bin

### ISSUE THREAD

**Issue by REmerald** [chaotic-aur/packages#2343]
### Package name

`neovim-nightly-bin`
https://aur.archlinux.org/packages/neovim-nightly-bin

### Build log

[neovim-nightly-bin.log](https://builds.garudalinux.org/repos/chaotic-aur/logs/neovim-nightly-bin.log)

### Latest build

`0.9.0+dev+1135+g419819b62-1`

### Latest version available

`0.9.0+dev+1265+g410dc3349-1`

### Have you tested if the package builds in a clean chroot?

- [X] Yes

### More information

Basically, the AUR packages pkgbuild just downloads the latest version from the github (it doesn't care about the version listed in aur), so it's our job to rebuild it every time there's an update on the github, [usually it happens daily](https://github.com/neovim/neovim/commits/nightly).

**Comment by Technetium1** [comment]
This is already in an hourly routine to check for updates. It's possible that a build failed for a while, and therefore we only served the last successful build. Definitely up to date now
`==> Finished making: neovim-nightly-bin 0.9.0+dev+1288+g4863ca6b8-1 (Mon 27 Mar 2023 04:49:36 AM CEST)`
https://github.com/chaotic-aur/packages/blob/fc94a4eb9e79a484309a5e9008dc61dd34c3c771/dragon-cluster/hourly.txt#L972-L973

**Comment by REmerald** [comment]
@Technetium1 , there's still a problem with either the hourly routine or the package itself, [there was an update 9 hours ago](https://github.com/neovim/neovim/releases/tag/nightly) and the chaotic-aur didn't see it:
```
[remorin@chuwawa:0 ~]$ paru
[sudo] password for root:
no new news
:: Synchronizing package databases...
 endeavouros is up to date
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
 chaotic-aur                                         2.4 MiB  2.23 MiB/s 00:01 [--------------------------------------------] 100%
 herecura is up to date
 maximbaz is up to date
 orhun is up to date
 arch4edu is up to date
:: Starting full system upgrade...
 there is nothing to do
:: Looking for AUR upgrades...
:: Looking for devel upgrades...
 there is nothing to do
[remorin@chuwawa:0 ~]$ paru -Sii chaotic-aur/neovim-nightly-bin
Repository      : chaotic-aur
Name            : neovim-nightly-bin
Version         : 0.9.0+dev+1288+g4863ca6b8-1
...
Packager        : Nico Jensch <dr460nf1r3@chaotic.cx>
Build Date      : Mon 27 Mar 2023 05:49:12 AM MSK
MD5 Sum         : 77e48709e7e088087b1fbb2d1d7c308a
SHA-256 Sum     : 8eff6bbb7667af5a8f894b81032be6c019047512dfaae305a9822697434f62df
Signatures      : 0706B90D37D9B881
```

**Comment by dr460nf1r3** [comment]
This is actually attributed to how the PKGBUILD works. The version on AUR never changes so repoctl assumes the package itself has not updated yet and doesn't start a build therefore.

**Comment by dr460nf1r3** [comment]
I typo'd in the commit, but this should fix it (forcing the redownload of the package) https://github.com/chaotic-aur/packages/commit/e02bbdf631d4f48d59910ff9ce51776e2bf98985

**Comment by dr460nf1r3** [comment]
I'm assuming the package list didn't get updated, as it worked right away when I triggered it manually. Let's observe this for a bit.

**Comment by REmerald** [comment]
Hmm, there was still no update yesterday (in chaotic-aur) and now there's an update, but still not to [the most recent version](https://github.com/neovim/neovim/releases/tag/nightly):
```
Repository      : chaotic-aur
Name            : neovim-nightly-bin
Version         : 0.9.0+dev+1307+g75e1b1d5c-1
...
Packager        : Nico Jensch <dr460nf1r3@chaotic.cx>
```
And judging by the `Packager` field, it was triggered manually, wasn't it?

**Comment by Technetium1** [comment]
@REmerald The only way to know that would be to look at the log: [neovim-nightly-bin.log](https://builds.garudalinux.org/repos/chaotic-aur/logs/neovim-nightly-bin.log). That's the expected packager for things built on dragon-cluster.

**Comment by REmerald** [comment]
@Technetium1 , actually, I'm not that familiar with the `makepkg`, so I can't tell it from the log.

Btw, it seems like, it fails to build?
```
==> ERROR: A package has already been built. (use -f to overwrite)
```
So, it seems like it fetches the newest version, it just fails to update the version itselt?

**Comment by Technetium1** [comment]
@REmerald that is expected happen when the latest version indicated in pkgbuild matches what is currently available in repo. It's at least indicating that the routine has run and checked the version. @dr460nf1r3 any ideas? 🤔

**Comment by REmerald** [comment]
Hmm, the good thing is that the update happens somewhat regularly:
```
[2023-04-02T13:34:31+0300] [ALPM] upgraded neovim-nightly-bin (0.9.0+dev+1301+g5bf2f4b3c-1 -> 0.9.0+dev+1307+g75e1b1d5c-1)
[2023-04-03T13:27:44+0300] [ALPM] upgraded neovim-nightly-bin (0.9.0+dev+1307+g75e1b1d5c-1 -> 0.9.0+dev+1312+g908494889-1)
[2023-04-05T01:30:19+0300] [ALPM] upgraded neovim-nightly-bin (0.9.0+dev+1312+g908494889-1 -> 0.9.0+dev+1319+g10baf8971-1)
[2023-04-06T12:02:15+0300] [ALPM] upgraded neovim-nightly-bin (0.9.0+dev+1319+g10baf8971-1 -> 0.9.0+dev+1338+g9e7426718-1)
```
Ofc, it's not ideal: [there was a release 3 hours ago](https://github.com/neovim/neovim/releases/tag/nightly), and `chaotic-aur` still uses previous version.

So, it seems like currently only the daily routine is working and not hourly?

**Comment by dr460nf1r3** [comment]
Once the timestamps commit is available at ours builders, we will know for sure!

**Comment by REmerald** [comment]
@Technetium1 @dr460nf1r3 , so the package has been doing great for the past 2+ months, it updates (almost?) daily (keep in mind, I mostly update my system once a day, inconsistently), and that's all that I care about.

It may be 1 version behind from github (honestly, I haven't checked it for these 2 months, and I won't check it now) during some periods in a day (or all the time?), but that's fine for me.  
You can monitor it using the timestamps if you want to debug it.

Should I close the issue now?

**Comment by xiota** [comment]
Out of curiosity, is there a need to yave both `neovim-git` and `neovim-nightly-bin`?

**Comment by REmerald** [comment]
@xiota, I think many ppl in Arch community are developers, so even if there's only a little fraction of ppl using the git version, that will still result in a lot of ppl, so lets just leave it how it is.

**Comment by xiota** [comment]
I'm wondering why bother with `neovim-nightly-bin`.  My understanding is it's preferred to avoid `-bin` packages on chaotic.

**Comment by Technetium1** [comment]
@xiota it's not the same, `-git` is actually building from commit, nightly builds once a day on nightly branch only https://github.com/neovim/neovim/blob/8376e8700b3d4f8ab96751612c3f4439a43b06f4/.github/workflows/release.yml#L4

**Comment by xiota** [comment]
Doesn't it end up being effectively the same though?  As if `-git` were being built once per day?

What about switching to `neovim-nightly`?

**Comment by Technetium1** [comment]
No it doesn't end up being the same; we build git hourly. The git version also builds from master, whereas nightly builds from nightly. If we were only doing a single daily build, then that would definitely be considered.

**Comment by REmerald** [comment]
If you guys prefer source packages over prebuilt binaries, you can try replacing `neovim-nightly-bin` with `neovim-nightly`

**Comment by REmerald** [comment]
@Technetium1, @dr460nf1r3, idk how I didn't notice it before, `neovim-nightly-bin` stopped updating 2 weeks ago...
```
Build Date      : Thu 22 Jun 2023 03:52:44 AM MSK
```

Hmm, maybe now it's time to switch to `neovim-nightly`? (If the information that `-bin` packages are less prefered than source is correct.)

**Comment by xiota** [comment]
Someone opened a deletion request against `neovim-nightly`, based on the premise that it's a duplicate of `neovim-git`.  Better to leave it alone to avoid having to revert later, at least until the request has been closed.

`neovim-nightly` might also have similar update issues as `neovim-nightly-bin`.

As far as source being preferred over binaries, I recall seeing it in a comment or message, but don't recall where it is or who wrote it.

**Comment by xiota** [comment]
I notice in the package list, `neovim-nightly-bin` is listed like a pkgbase, but on AUR it looks like a normal package.  Does that help with rebuilding?

https://github.com/chaotic-aur/packages/blob/2e6967bd62d52ac19d507bc9aef3a19e3410f859/garuda-cluster/hourly.1.txt#L565-L566

**Comment by dr460nf1r3** [comment]
> Note: Last successful rebuild was Aug 2. Last build attempt was Aug 5 (today), but with cgroup error.

These still hunt us :( It's super random and there is no real indicator as to why it happens after investigating it for some time now. cc @JustTNE

**Comment by xiota** [comment]
`neovim-nightly-bin` last built 2023-09-14.  This issue seems to be resolved now.

**Comment by REmerald** [comment]
@xiota, version didn't change, it should be similar to `0.10.0-dev-1720+g52d738826`. See the first link in my previous comment.

**Comment by xiota** [comment]
Rebuilt again.  The way the package is constructed, builds can lag releases by a day.  I will comment on aurweb to tell maintainer how to fix it.

**Comment by REmerald** [comment]
@xiota, thanks, it fixed it.
```
[2023-12-01T01:24:38+0300] [ALPM] upgraded neovim-nightly-bin (0.10.0+dev+64b53b7-1 -> 0.10.0+dev+1720+g52d738826-1)
```

I don't think the maintainer reads the AUR comments, the best way would be to ping them in [the issue I created](https://github.com/eclairevoyant/aur-packages/issues/10) or create a new one (but you will still need to ping to get an answer).

**Comment by xiota** [comment]
I am switching this package to build nightly only.  That should reduce issues with out of date builds.  Until the version mismatch issue is resolved upstream, please don't report outdated until they're older than two days old.
